ci/build.sh

```diff
@@ -26,52 +51,60 @@ function bundle_fluentd_plugins() {
     exit 1
   fi
 
-  for i in ./fluent-plugin-*/ ; do
-    if [[ -d "${i}" ]]; then
-    (
-      cd "${i}" || exit 1
-      local plugin_name
-      plugin_name="$(basename "${i}")"
-      # Strip everything after "-" (longest match) to avoid gem prerelease behavior
-      local gem_version="${version%%-*}"
-      echo "Building gem ${plugin_name} version ${gem_version} in $(pwd) ..."
-      sed -i.bak "s/0.0.0/${gem_version}/g" ./"${plugin_name}".gemspec
-      rm -f ./"${plugin_name}".gemspec.bak
-
-      echo "Install bundler..."
-      bundle install
-
-      echo "Run unit tests..."
-      bundle exec rake
-
-      echo "Build gem ${plugin_name} ${gem_version}..."
-      gem build "${plugin_name}"
-      mv ./*.gem ../deploy/docker/gems
-    )
-    fi
-  done
+  find . -name 'fluent-plugin-*' -type 'd' -print |
+    while read -r line; do
+      # Run tests in their own context
+      (bundle_fluentd_plugin "$(basename "${line}")" "${version}") || exit 1
+    done
 }
 
-# Set up Github
-if [ -n "$GITHUB_TOKEN" ]; then
+function set_up_github() {
+  local token="${1}"
+  local branch="${2}"
+
   git config --global user.email "travis@travis-ci.org"
   git config --global user.name "Travis CI"
-  git remote add origin-repo "https://${GITHUB_TOKEN}@github.com/SumoLogic/sumologic-kubernetes-collection.git" > /dev/null 2>&1
+  git remote add origin-repo "https://${token}@github.com/SumoLogic/sumologic-kubernetes-collection.git" > /dev/null 2>&1
   git fetch --unshallow origin-repo
 
-  readonly branch_to_checkout="$(get_branch_to_checkout)"
-  echo "Checking out the ${branch_to_checkout} branch..."
-  git checkout "${branch_to_checkout}"
-fi
+  readonly branch="$(get_branch_to_checkout)"
+  echo "Checking out the ${branch} branch..."
+  git checkout "${branch}"
+}
 
-## check the build script with shellcheck
-## TODO: the "|| true" prevents the build from failing on shellcheck errors - to be removed
-echo "Checking the build script with shellcheck..."
-shellcheck ci/build.sh || true
+function push_docker_image() {
+  local version="$1"
+
+  echo "Tagging docker image $DOCKER_TAG:local with $DOCKER_TAG:$version..."
+  docker tag "$DOCKER_TAG:local" "$DOCKER_TAG:$version"
+  echo "Pushing docker image $DOCKER_TAG:$version..."
+  echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  docker push "$DOCKER_TAG:$version"
+}
+
+function push_helm_chart() {
+  local version="$1"
+
+  echo "Pushing new Helm Chart release $version"
+  set -x
```

Comment:
Do we want to keep it?

Reply:
Yeah, that's very helpful when debugging helm release issues.